zeebe/gateway-grpc/src/main/java/io/camunda/zeebe/gateway/interceptors/impl/InterceptorRepository.java

```diff
@@ -31,60 +27,59 @@ public final class InterceptorRepository {
   private static final Logger LOGGER = Loggers.GATEWAY_LOGGER;
 
   private final ExternalJarRepository jarRepository;
-  private final SortedMap<InterceptorId, Class<? extends ServerInterceptor>> interceptors;
+  private final List<InterceptorElement> interceptors;
```

Comment:
💭 Using a `LinkedHashMap`, you would directly keep the insertion order when retrieving all elements and have map-like access, no need to search through the list to find a specific element. I think it doesn't matter much since we probably always deal with small lists of interceptors anyway. But just as food for thought.

Reply:
That's a nice suggestion, I'll re-work it. Thanks!